src/wallet/wallet_para_helpers.h

```diff
@@ -65,4 +107,34 @@ class sthd_blk_fetcher
 	uint32_t m_rpc_version;
 };
 
+//Since network is not exactly a CPU intensive task, one size will fit all
+class mthd_blk_fetcher
+{
+public:
+	mthd_blk_fetcher(size_t max_q_cap, const std::string &daemon_address, boost::optional<epee::net_utils::http::login> daemon_login);
+	void start_fetching(std::list<crypto::hash> &&short_chain_history);
+	bool get_batch(blk_batch &output);
+	void join_thread();
+	
+private:
+	bool is_finished();
+	void fetch_thd_main();
+	
+	sthd_blk_fetcher fetcher;
+		
+	std::thread thd;
+	std::atomic<bool> thd_exit;
+	std::atomic<bool> thd_run;
+
+	size_t max_q_cap;
+	std::mutex q_cvm;
+	std::condition_variable cv_q_nempty;
```

Comment:
Ok. I will change the stuff that you pointed out to boost, you seem to use std::atomic without any troubles in other places, I assume that is ok?